From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 2 May 2013 12:03:18 +0200
Subject: virthreadpthread: don't try to cast pthread_t to void*

This fixes the build on kFreeBSD that otherwise fails with:

util/virthreadpthread.c: In function 'virThreadSelfID':
util/virthreadpthread.c:222:27: error: cast from function call of type 'pthread_t' to non-matching type 'void *' [-Werror=bad-function-cast]
---
 src/util/virthreadpthread.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/virthreadpthread.c b/src/util/virthreadpthread.c
index b42b333..8d5e197 100644
--- a/src/util/virthreadpthread.c
+++ b/src/util/virthreadpthread.c
@@ -219,7 +219,7 @@ int virThreadSelfID(void)
     tid = syscall(SYS_gettid);
     return (int)tid;
 #else
-    return (int)(intptr_t)(void *)pthread_self();
+    return (int)(intptr_t)pthread_self();
 #endif
 }
 
